### PR TITLE
Save syscall table and binary log in one file

### PR DESCRIPTION
**Pull request by ldorau**
This change is [Reviewable](https://reviewable.io/reviews/pmem/vltrace/20)

### CODE REVIEW

File: src/ebpf_syscalls.c [hunk 1]
```diff
@@ -700,23 +700,17 @@ print_syscalls_table(FILE *f)
  * dump_syscalls_table -- dump the table of syscalls
  */
 int
-dump_syscalls_table(const char *path)
+dump_syscalls_table(FILE *file)
 {
 	int size = sizeof(struct syscall_descriptor);
-	int ret = 0;
+	int count = SC_TBL_SIZE;
```
Comment (marcinslusarz) [inline src/ebpf_syscalls.c]: We are dealing with on-media format, so I think we should use fixed size types (uint32_t here) or verify its size using COMPILE_ERROR_ON. Now I see syscall_descriptor has the same problem...

Comment (ldorau) [reply src/ebpf_syscalls.c]: Done (syscall_descriptor will be fixed in a separate PR)

**Comment by marcinslusarz** [comment]
Reviewed 3 of 5 files at r1, 1 of 1 files at r2.
Review status: 4 of 5 files reviewed at latest revision, 1 unresolved discussion.

---

*[test/test-match, line 118 at r2](https://reviewable.io:443/reviews/pmem/vltrace/20#-Kq2xSgjHzV0Ac7YXWSa:-Kq2xSgjHzV0Ac7YXWSb:bbyskov) ([raw file](https://github.com/pmem/vltrace/blob/489057007b0e148ae7d1a29d9996e45593b973d6/test/test-match#L118)):*
> ```
> 	# binary output format
> 	echo "$ $BIN2TXT -t syscalls_table.dat -b $OUT_VLT > $OUT_TXT"
> 	$BIN2TXT -t syscalls_table.dat -b $OUT_VLT > $OUT_TXT
> ```

Do we need to remove -t option from bin2txt?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/vltrace/20)*

**Comment by ldorau** [comment]
Review status: 4 of 5 files reviewed at latest revision, 1 unresolved discussion.

---

*[test/test-match, line 118 at r2](https://reviewable.io:443/reviews/pmem/vltrace/20#-Kq2xSgjHzV0Ac7YXWSa:-Kq33ztwvGtq-kx4lDhs:b-p4fw54) ([raw file](https://github.com/pmem/vltrace/blob/489057007b0e148ae7d1a29d9996e45593b973d6/test/test-match#L118)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

Do we need to remove -t option from bin2txt?
</blockquote></details>

Yes, it will be removed in a separate commit

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/vltrace/20)*

**Comment by ldorau** [comment]
Will be continued in PR https://github.com/pmem/vltrace/pull/22

**Pull request closed without merging**